From 923370a15dc294c3e754437aba18552930115a91 Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Mon, 15 Dec 2014 17:00:54 +0800
Subject: [PATCH] vdsm.spec.in: add with_hooks.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 vdsm.spec.in | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/vdsm.spec.in b/vdsm.spec.in
index a1afac0..4b98574 100644
--- a/vdsm.spec.in
+++ b/vdsm.spec.in
@@ -15,6 +15,9 @@
 # Fedora official build
 %global fedora_koji_build 0
 
+# Also build hooks
+%global with_hooks 1
+
 # Default to skipping autoreconf.  Distros can change just this one line
 # (or provide a command-line override) if they backport any patches that
 # touch configure.ac or Makefile.am.
@@ -1537,6 +1540,12 @@ exit 0
 %endif
 
 %changelog
+* Mon Dec 15 2014 zhaochao <chao.zhao@eayun.com> - 4.16.7-4
+- add with_hooks to SPEC.
+
+* Mon Nov 24 2014 huntxu <mhuntxu@gmail.com> - 4.16.7-3
+- hostusb passthrough.
+
 * Sun Oct 13 2013 Yaniv Bronhaim <ybronhei@redhat.com> - 4.13.0
 - Removing vdsm-python-cpopen from the spec
 - Adding dependency on formal cpopen package
-- 
1.8.3.1

