From fe3ad6b27d82e2cf3c69d643259d8aad00b06d3d Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Sat, 20 Dec 2014 20:32:38 +0800
Subject: [PATCH 6/6] update SPEC file.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 vdsm.spec.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/vdsm.spec.in b/vdsm.spec.in
index 5dcb1e8..a931f6e 100644
--- a/vdsm.spec.in
+++ b/vdsm.spec.in
@@ -1528,6 +1528,11 @@ exit 0
 %endif
 
 %changelog
+* Sat Dec 20 2014 zhaochao <chao.zhao@eayun.com> - 4.16.7-8
+- merge upstream patch for fix image device seliunx labels
+  using chown not udev.
+- change default mgmt bridge name to eayunosmgmt.
+
 * Mon Dec 15 2014 zhaochao <chao.zhao@eayun.com> - 4.16.7-5
 - add a new OSName for EayunOS in vdsm/caps.
 
-- 
1.8.3.1

