From cef98031b7915f61aa078b3001802f22bafe7c99 Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Sat, 20 Dec 2014 19:16:02 +0800
Subject: [PATCH 2/3] add 'eayunosmgmt' as a default bridge name.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 src/engine_page.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/engine_page.py b/src/engine_page.py
index ae3ef9c..9cc77ee 100644
--- a/src/engine_page.py
+++ b/src/engine_page.py
@@ -89,7 +89,7 @@ def sync_mgmt():
         networks = cli.getVdsCapabilities()['info']['networks']
 
         for net in networks:
-            if net in ('ovirtmgmt', 'rhevm'):
+            if net in ('ovirtmgmt', 'rhevm', 'eayunosmgmt'):
                 if 'bridge' in networks[net]:
                     mgmtIface = [networks[net]['bridge']]
                 else:
-- 
1.8.3.1

