From 7be208f16e0090a267e2af7221283a92f16daed8 Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Sat, 20 Dec 2014 19:20:36 +0800
Subject: [PATCH 3/3] fix default values for Entries in UI.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 src/engine_page.py | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/engine_page.py b/src/engine_page.py
index 9cc77ee..13debcc 100644
--- a/src/engine_page.py
+++ b/src/engine_page.py
@@ -150,8 +150,16 @@ class Plugin(plugins.NodePlugin):
     def model(self):
         cfg = VDSM().retrieve()
         model = {
-            "vdsm_cfg.address": cfg["server"] or "",
-            "vdsm_cfg.port": cfg["port"] or "443",
+            "vdsm_cfg.address": cfg["server"]
+                if cfg["server"] and
+                   cfg["server"].upper() != "None".upper()
+                else
+                   "",
+            "vdsm_cfg.port": cfg["port"]
+                if cfg["port"] and
+                   cfg["port"].upper() != "None".upper()
+                else
+                   "443",
             "vdsm_cfg.cert": "Verified"
             if utils.fs.Config().exists(cfg["cert_path"]) else "N/A",
             "vdsm_cfg.password": "",
-- 
1.8.3.1

