From 6f957ef6bf8445c913edf8aed1dd76b9501114c7 Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Sat, 20 Dec 2014 19:41:55 +0800
Subject: [PATCH] add the first i18n modification.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 src/engine_page.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/engine_page.py b/src/engine_page.py
index 13debcc..cd0bce0 100644
--- a/src/engine_page.py
+++ b/src/engine_page.py
@@ -142,7 +142,7 @@ class Plugin(plugins.NodePlugin):
         sync_mgmt()
 
     def name(self):
-        return config.engine_name
+        return _(config.engine_name)
 
     def rank(self):
         return 100
-- 
1.8.3.1

