From 36dc15e3effe7ca714b44f355fdc29d486be67dd Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Sat, 20 Dec 2014 20:52:21 +0800
Subject: [PATCH] change rank to 114.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 src/engine_page.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/engine_page.py b/src/engine_page.py
index cd0bce0..02e230e 100644
--- a/src/engine_page.py
+++ b/src/engine_page.py
@@ -145,7 +145,7 @@ class Plugin(plugins.NodePlugin):
         return _(config.engine_name)
 
     def rank(self):
-        return 100
+        return 114
 
     def model(self):
         cfg = VDSM().retrieve()
-- 
1.8.3.1

