From 94b68fcacf825e6785a6270c8cb3be54fbf81ad6 Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Wed, 26 Nov 2014 11:01:20 +0800
Subject: [PATCH 2/2] add support for local images.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 src/hosted_engine_page.py | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/src/hosted_engine_page.py b/src/hosted_engine_page.py
index 376fce6..1a64c38 100644
--- a/src/hosted_engine_page.py
+++ b/src/hosted_engine_page.py
@@ -34,6 +34,7 @@ import time
 
 VM_CONF_PATH = "/etc/ovirt-hosted-engine/vm.conf"
 HOSTED_ENGINE_SETUP_DIR = "/data/ovirt-hosted-engine-setup"
+HOSTED_ENGINE_SETUP_TEMPDIR = HOSTED_ENGINE_SETUP_DIR + "/tempdir"
 """
 Configure Hosted Engine
 """
@@ -48,13 +49,19 @@ def get_hosted_cfg(imagepath, pxe):
     txt = "[environment:default]\n"
     if imagepath.endswith(".iso"):
         boot = "cdrom"
-        txt += "OVEHOSTED_VM/vmCDRom=str:%s\n" % \
-               os.path.join(HOSTED_ENGINE_SETUP_DIR, imagepath)
+        if imagepath.startswith("/"):
+            txt += "OVEHOSTED_VM/vmCDRom=str:%s\n" % imagepath
+        else:
+            txt += "OVEHOSTED_VM/vmCDRom=str:%s\n" % \
+                    os.path.join(HOSTED_ENGINE_SETUP_DIR, imagepath)
     elif imagepath.endswith(".gz"):
         boot = "disk"
         ova = True
-        ova_path = os.path.join(HOSTED_ENGINE_SETUP_DIR,
-                                os.path.basename(imagepath))
+        if imagepath.startswith("/"):
+            ova_path = imagepath
+        else:
+            ova_path = os.path.join(HOSTED_ENGINE_SETUP_DIR,
+                    os.path.basename(imagepath))
     elif pxe:
         boot = "pxe"
     if boot:
@@ -63,6 +70,7 @@ def get_hosted_cfg(imagepath, pxe):
         txt += "OVEHOSTED_VM/vmBoot=none:None\n"
     if ova:
         txt += "OVEHOSTED_VM/ovfArchive=str:%s\n" % ova_path
+        txt += "OVEHOSTED_CORE/tempDir=str:%s\n" % HOSTED_ENGINE_SETUP_TEMPDIR
     else:
         txt += "OVEHOSTED_VM/ovfArchive=none:None\n"
     return txt
@@ -116,7 +124,7 @@ class Plugin(plugins.NodePlugin):
         return self._model
 
     def validators(self):
-        return {"hosted_engine.diskpath": valid.Empty() | valid.URL()}
+        return {"hosted_engine.diskpath": valid.Empty() | valid.Text()}
 
     def ui_content(self):
         ws = [ui.Header("header[0]", "Hosted Engine Setup"),
@@ -130,7 +138,7 @@ class Plugin(plugins.NodePlugin):
 
               ui.Divider("divider[0]"),
               ui.Entry("hosted_engine.diskpath",
-                       "Engine ISO/OVA URL for download:"),
+                       "Engine ISO/OVA Location:"),
               ui.Divider("divider[1]"),
               ui.Checkbox("hosted_engine.pxe", "PXE Boot Engine VM")
               ]
-- 
1.8.3.1

