From 00a5e30e88bea7097954c34051c396f055bbb5ad Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Sat, 20 Dec 2014 20:45:45 +0800
Subject: [PATCH] add the first i18n mod, change rank to 115.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 src/hosted_engine_page.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/hosted_engine_page.py b/src/hosted_engine_page.py
index f258426..0aa7ab6 100755
--- a/src/hosted_engine_page.py
+++ b/src/hosted_engine_page.py
@@ -54,10 +54,10 @@ class Plugin(plugins.NodePlugin):
         super(Plugin, self).__init__(application)
 
     def name(self):
-        return "Hosted Engine"
+        return _("Hosted Engine")
 
     def rank(self):
-        return 110
+        return 115
 
     def model(self):
         configured = os.path.exists(self.VM_CONF_PATH)
-- 
1.8.3.1

