From 0d7ef214e2b6b87a2260793b727af4473bf1cc1b Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Sat, 20 Dec 2014 21:25:35 +0800
Subject: [PATCH 2/4] fix default iscsi iname substituion.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 src/ovirtnode/iscsi.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ovirtnode/iscsi.py b/src/ovirtnode/iscsi.py
index 9dbbaf0..3666b3a 100755
--- a/src/ovirtnode/iscsi.py
+++ b/src/ovirtnode/iscsi.py
@@ -50,6 +50,10 @@ def iscsi_auto():
         iscsi_iqn_cmd = _functions.subprocess_closefds("/sbin/iscsi-iname", \
                                                        stdout=subprocess.PIPE)
         iscsi_iqn, err = iscsi_iqn_cmd.communicate()
+        iscsi_iqn = iscsi_iqn.replace(
+                "1994-05.com.redhat",
+                "2011-06.com.eayun"
+                )
         set_iscsi_initiator(iscsi_iqn.strip())
     else:
         set_iscsi_initiator(_functions.OVIRT_VARS["OVIRT_ISCSI_NAME"])
-- 
1.8.3.1

