From 52a39ecbf81b66436b74a8cd0119cbe1f5c1b97e Mon Sep 17 00:00:00 2001
From: walteryang <walteryang47@gmail.com>
Date: Tue, 31 Mar 2015 17:32:58 +0800
Subject: [PATCH] Add support for centos senario when efi boot

Signed-off-by: walteryang <walteryang47@gmail.com>
---
 src/ovirtnode/install.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/ovirtnode/install.py b/src/ovirtnode/install.py
index abf9849..c8c2e80 100755
--- a/src/ovirtnode/install.py
+++ b/src/ovirtnode/install.py
@@ -96,11 +96,15 @@ class Install:
 
         if os.path.exists("/boot/efi/EFI/fedora"):
             self.efi_dir_name = "fedora"
+        elif os.path.exists("/boot/efi/EFI/centos"):
+            self.efi_dir_name = "centos"
         else:
             self.efi_dir_name = "redhat"
         if _functions.is_efi_boot():
             if self.efi_dir_name == "fedora":
                 self.grub_config_file = "/liveos/efi/EFI/fedora/grub.cfg"
+            elif self.efi_dir_name == "centos":
+                self.grub_config_file = "/liveos/efi/EFI/centos/grub.cfg"
             else:
                 self.grub_config_file = "/liveos/efi/EFI/redhat/grub.conf"
 
@@ -360,6 +364,8 @@ initrd /initrd0.img
             _functions.mount_efi(target="/liveos")
             if self.efi_dir_name == "fedora":
                 grub_config_file = "/liveos/EFI/fedora/grub.cfg"
+            elif self.efi_dir_name == "centos":
+                grub_config_file = "/liveos/EFI/centos/grub.cfg"
             else:
                 grub_config_file = "/liveos/EFI/redhat/grub.conf"
         grub_config_file_exists = grub_config_file is not None \
@@ -520,6 +526,11 @@ initrd /initrd0.img
                                              ("\\EFI\\%s\\grubx64.efi" %
                                               self.efi_dir_name),
                                              efi_disk)
+                elif self.efi_dir_name == "centos":
+                    _functions.add_efi_entry(_functions.PRODUCT_SHORT,
+                                             ("\\EFI\\%s\\grubx64.efi" %
+                                              self.efi_dir_name),
+                                             efi_disk)
                 else:
                     if os.path.isfile("/liveos/efi/EFI/redhat/grubx64.efi"):
                         _functions.add_efi_entry(_functions.PRODUCT_SHORT,
-- 
1.9.3

